// spinners-react/lib/cjs/SpinnerRoundFilled.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var styleInject_es = require('./style-inject.es-53b0d2f9.js');
var React = _interopDefault(require('react'));

var css_248z = "@keyframes spinners-react-round-filled-outer{40%,60%{transform:scale(0)}}@keyframes spinners-react-round-filled-center{30%,70%{transform:scale(0)}}@keyframes spinners-react-round-filled-inner{20%,80%{transform:scale(0)}}";
styleInject_es.styleInject(css_248z);

var SpinnerRoundFilled = function (_a) {
    var color = _a.color, enabled = _a.enabled, size = _a.size, speed = _a.speed, still = _a.still, thickness = _a.thickness, style = _a.style, svgProps = styleInject_es.__rest(_a, ["color", "enabled", "size", "speed", "still", "thickness", "style"]);
    var svgStyle = styleInject_es.__assign({ color: color, overflow: 'visible', width: styleInject_es.normalizeSize(size) }, style);
    var animations = [
        { r: 4 },
        {
            name: 'spinners-react-round-filled-inner',
            r: 12.66,
        },
        {
            name: 'spinners-react-round-filled-center',
            r: 20.32,
        },
        {
            name: 'spinners-react-round-filled-outer',
            r: 27.5,
        },
    ];
    if (!enabled)
        return null;
    return (React.createElement("svg", styleInject_es.__assign({ fill: "none" }, svgProps, { style: svgStyle, viewBox: "0 0 66 66" }), animations.map(function (animation) { return (React.createElement("circle", { key: animation.name || 'still', cx: "33", cy: "33", fill: "currentColor", r: animation.r * (animation.name ? (thickness / 100) : 1), style: {
            opacity: animation.name ? 0.25 : 1,
            transformOrigin: 'center',
            animation: (!animation.name || still)
                ? 'none'
                : animation.name + " " + 140 / speed + "s ease-in-out infinite",
        } })); })));
};
SpinnerRoundFilled.defaultProps = styleInject_es.defaultProps;

exports.SpinnerRoundFilled = SpinnerRoundFilled;
